react/components/MaybeAuth.tsx

```diff
@@ -57,25 +58,32 @@ export default class MaybeAuth extends PureComponent<Props, State> {
     return null
   }
 
-  private onUpdate() {
+  private async onUpdate() {
     if (this.isAuthenticatedPage()) {
-      fetch(AUTH_STORE_URL, { credentials: 'same-origin' })
-      .then(response => response.json())
-      .then(({ authenticated }) => {
+      const promiseSession = fetch(API_SESSION_URL, { credentials: 'same-origin' })
+        .then(response => response.json())
+        .catch(() => null)
+      const promiseAuth = fetch(AUTH_STORE_URL, { credentials: 'same-origin' })
+        .then(response => response.json())
+        .catch(() => null)
+      const [sessionRes, authRes] = await Promise.all([
+        promiseSession, promiseAuth
+      ])
+      if (
+        (sessionRes && sessionRes.namespaces.profile.isAuthenticated.value !== 'False') ||
```

Comment:
Create a func `convertToBool`, and return `toLower(sessionRes.namespaces.profile.isAuthenticated.value) === 'true' `